res/controllers/DJControl-Compact-scripts.js

```diff
@@ -0,0 +1,113 @@
+// Mode button does give out midi, but the selection itself is done inside the
+// controller -- no control over which mode is selected for instance.
```

Comment:
to select pad modes, you hold the mod button and then press one of the pads.  That functionality is in the fimeware of the controller and can't be changed.  And then yes, the pads send different midi signals depending on what modes are selected.  All buttons also send different messages when shift is held, so I don't even have to program the shift button in js!  (I'll be filling out all this information in comments in a subsequent commit)